IntegrationTests/Services/AWSS3IntegrationTests/SSOCredentialsProviderTests.swift

```diff
@@ -0,0 +1,105 @@
+ * 5. Retrieve username, password, and SSO start URL of the IAM Identity Store user. Configure SSO config using these values (aws sso configure).
+ * 6. Create SSO token using AWS CLI (aws sso login --profile <profile-name-set-during-aws-sso-configure>).
+ * 7. Run the test.
+ *
+ * Note: Enabling AWS organization and creating IAM identity store have no exposed API as of 9/17/2023 for programmatic activation / creation.
+ *
+ * Step 1, 3, 4, 5 have to be done only once per account.
+ */
+class SSOCredentialsProviderTests : XCTestCase {
+    var client: S3Client!
+    var ssoClient: SSOAdminClient!
+    // Change this region string to the region where AWS SSO instance is in.
+    let region = "us-west-2"
+    
+    private var iamIdentityCenterInstanceArn: String = ""
+    private let permissionSetName = "SSOCredProvIntegTestPermSet"
+    private var permissionSetArn: String = ""
+    private let awsReadOnlyPolicy = "arn:aws:iam::aws:policy/ReadOnlyAccess"
+    
+        
+    override func setUp() async throws {
+        // Use default credentials provider chain for setup
+        ssoClient = try SSOAdminClient(region: "us-west-2")
+        try await createPermissionSetIfNeeded()
+        
+        // Create a S3 client that uses SSO credentials provider
+        try await setUpClient()
+    }
+
+    // The test calls listBuckets() and forces S3Client to use SSOCredentialsProvider
+    func test_listBuckets() async throws {
+        _ = try await client.listBuckets(input: ListBucketsInput())
+    }
+    
+    override func tearDown() async throws {
+        // No action required.
+    }
+    
+    /* HELPER METHODS */
+    private func createPermissionSetIfNeeded() async throws {
+        // Get IAM identity center instanceArn
+        let listInstancesOutput = try await ssoClient.listInstances(input: ListInstancesInput())
+        iamIdentityCenterInstanceArn = listInstancesOutput.instances![0].instanceArn!
+        
+        do {
+            // Create permission set and save its ARN
+            let createPermissionSetOutput = try await ssoClient.createPermissionSet(input: CreatePermissionSetInput(
+                description: "Permission set for testing SSO credentials provider.",
+                instanceArn: iamIdentityCenterInstanceArn,
+                name: permissionSetName
+            ))
+            permissionSetArn = createPermissionSetOutput.permissionSet!.permissionSetArn!
+            
+            
+            // Attach ReadOnly AWS-managed policy to the created permission set
+            let _ = try await ssoClient.attachManagedPolicyToPermissionSet(input: AttachManagedPolicyToPermissionSetInput(
+                instanceArn: iamIdentityCenterInstanceArn,
+                managedPolicyArn: awsReadOnlyPolicy,
+                permissionSetArn: permissionSetArn))
+        } catch let error as AWSSSOAdmin.ConflictException {
+            if error.message == "PermissionSet with name \(permissionSetName) already exists." {
```

Comment:
Assume error messages can change at any time.  Is there a more reliable way to verify this error condition?

Reply:
Currently there's no other way to check the specific cause for the error other than checking the message contained int he error. So as a workaround I now look for smaller substring using `contains()` to reduce dependency on the entire error message string staying the same. Now it only checks that `\(permissionSetName) already exists` is part of the error message.